### PR TITLE
Add dotnet core implementation in C#

### PR DESCRIPTION
**Pull request by mdonatas**
- Implementation is Java variant converted to C#
- Tests run all test cases from `test/round_half_up`

### CODE REVIEW

File: dotnetcore/test/FlexiblePolylineEncoder.Tests/FlexiblePolylineEncoderTests.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
+using System;
```
Comment (lene) [inline dotnetcore/test/FlexiblePolylineEncoder.Tests/FlexiblePolylineEncoderTests.cs:1]: I see that here you only implemented the smoke tests from the Java implementation. To ensure the algorithm is correct, the other tests are necessary too. Please implement them as well - and also have separate smoke tests for encoding and decoding, such as `encodingSmokeTest()` and `decodingSmokeTest()` in `PolylineEncoderDecoderTest.java`.

**Comment by TonyLugg** [comment]
Can someone finish this off? :)

**Comment by dhippo78** [comment]
This version of the [decoder](https://github.com/dhippo78/flexible-polyline/commit/43520ad22d8912fbcb80a77a532ef3bbfbbde171) I converted, I use without any issues:

**Comment by mlugg01** [comment]
> This version of the [decoder](https://github.com/dhippo78/flexible-polyline/commit/43520ad22d8912fbcb80a77a532ef3bbfbbde171) I converted, I use without any issues:

I can't get this to compile - is it Xamarin specific?

**Comment by lene** [comment]
Closing since PR #31 has been merged now.

**Pull request closed without merging**